### PR TITLE
Allow default_tags to be set by environment variables

### LINKED ISSUE #33255: [Enhancement]: default_tags from environment variables
### Description

I would like to be able to configure the `default_tags` of the AWS provider using environment variables.

This would be useful, for example, in Terraform Cloud/Enterprise. If I have a bunch of workspaces organized under a project, I could attach a variable set to the project with an environment variable that would configure a tag like `Owner=my-team` across the entire set of AWS resources under that project.

### Affected Resource(s) and/or Data Source(s)

* aws provider

### Potential Terraform Configuration

```terraform
# Read default_tags from environment variables like:
# TF_AWS_DEFAULT_TAGS_<tag_name>=<tag_value>

provider "aws" {
  # Suppose the following environment variables are set:
  # TF_AWS_DEFAULT_TAGS_Owner=my-team
  # TF_AWS_DEFAULT_TAGS_Application=my-app

  default_tags {
    tags = {
      Application = "foobar" # Explicit configuration should override the env vars.
    }
  }

  # The resulting tags map should be `{ Owner = "my-team", Application = "foobar" }`.
}
```

### References

Pull request to implement this change: https://github.com/hashicorp/terraform-provider-aws/pull/33339

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by jtdoepke**
### Description

This commit has the provider consume `default_tags` from environment variables with the form `TF_AWS_DEFAULT_TAGS_<tag_name>=<tag_value>`. If the same tag is defined in both an environment variable and the Terraform config, the Terraform takes precedence.

### Relations

Closes #33255

### Output from Acceptance Testing

```console
% make testacc TESTS=TestAccProvider_DefaultTagsTags_env_vars PKG_NAME=internal/provider

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/provider/... -v -count 1 -parallel 20 -run='TestAccProvider_DefaultTagsTags_env_vars'  -timeout 180m
?       github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider        [no test files]
=== RUN   TestAccProvider_DefaultTagsTags_env_vars
--- PASS: TestAccProvider_DefaultTagsTags_env_vars (20.22s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/provider   20.660s
```

### CODE REVIEW

File: internal/tags/key_value_tags.go [hunk 1]
```diff
@@ -28,6 +28,9 @@ const (
 	ElasticbeanstalkTagKeyPrefix                = `elasticbeanstalk:`
 	NameTagKey                                  = `Name`
 	ServerlessApplicationRepositoryTagKeyPrefix = `serverlessrepo:`
+
+	// Environment variables prefixed with this string will be treated as default_tags.
+	DefaultTagsEnvVarPrefix = "TF_AWS_DEFAULT_VARS_"
```
Comment (brandon-berg) [inline internal/tags/key_value_tags.go:33]: Shouldn't this be "TF_AWS_DEFAULT_TAGS_"?

Comment (jtdoepke) [reply internal/tags/key_value_tags.go:33]: Good catch!

**Comment by bcbrockway** [comment]
Just wondering what's left to get this over the line? We could really use this functionality.

**Review by jar-b — APPROVED** [review]
LGTM 🚀

**Comment by jar-b** [comment]
Thanks for your contribution, @jtdoepke! 👍

**Pull request merged**